lib/test/Test/Pos/Ssc/Toss/BaseSpec.hs

```diff
@@ -616,6 +622,33 @@ checksBadCertsPayload (GoodPayload epoch sgs certsMap mrs) pk cert =
         not (vcVssKey cert `elem` allVssKeys))
        ==> res1 .&&. res2 .&&. res3 .&&. res4
 
+-- [CT-99]: Test that no two stakeholders can have same active VSS keys
+checksBijectiveCertsPayload :: HasConfiguration => GoodCertsPayload -> VssCertData -> Property
+checksBijectiveCertsPayload (GoodPayload _ _ vssCertsMap _) _ = do
+    -- 1. A VSS payload with two certs that have equal VSS keys should not be
+    -- deserializable (with decode).
+
+    -- insert same certificate with different ids
+    let certsMap = getVssCertificatesMap vssCertsMap
+    let x@(_, cert1) : (sid2, _) : rest = HM.toList certsMap
+    let newCerts = UnsafeVssCertificatesMap $ HM.fromList $ x : (sid2, cert1) : rest
+
+    -- check that certificate duplication throws error on decoding stage
+    let encodedCerts = encode newCerts
+    let decodedCerts = deserialiseFromBytes (decode @VssCertificatesMap)
+                                            (toLazyByteString encodedCerts)
+    let check1 = HM.size certsMap >= 2 ==> isLeft decodedCerts
```

Comment:
1. Why do you need `GoodPayload` for this test and not simply `VssCertificatesMap`?
2. `UnsafeVssCertificatesMap $ HM.fromList $ x : (sid2, cert1) : rest` is bad, because it violates two invariants of `VssCertificatesMap`:
* There are two same `vcVssKey` ← that's what we want to violate and get an error.
* `vcSigningKey` for one entry is not consistent with corresponding key in the map (i. e. `sid2 /= addressHash (vsSigningKey cert1)`.

So it's possible that we don't check the first condition and `decodedCerts` will still be `Left` and the test will pass when it shouldn't.
In order to fix it you should do the following:
1. Generate two different `SecretKey`s.
2. Generate 1 `VssPublicKey`.
3. Use `mkVssCertificate` with different `SecretKey`s and same `VssPublicKey` (epoch doesn't matter, you can always pass `0`, for example).
4. Take an arbitrary (valid) `VssCertificatesMap` and put these two certificates there (keys should be `addressHash (vsSigningKey cert1)`).